app/models/smart_proxy.rb

```diff
@@ -97,7 +97,7 @@ def has_capability?(feature, capability)
   end
 
   def setting(feature, setting)
-    smart_proxy_features.find_by(:feature_id => Feature.find_by(:name => feature)).try(:settings).try(:[], setting)
+    smart_proxy_features.find { |spf| spf.feature_id == Feature.find_by(:name => feature).id }.try(:settings).try(:[], setting)
```

Comment:
Most of this is the same as `capabilities` a bit above. Perhaps those two should share a common helper?

I also wonder if here the spf doesn't also have a name property already that you can match to, rather than the `Feature.find_by(:name)`

Reply:
updated to use feature_name on spf and to do similar with capabilities.  I don't know that adding a helper method is worthwhile in this case, but am open to adding it